x-pack/plugins/index_management/public/application/sections/home/data_stream_list/edit_data_retention_modal/edit_data_retention_modal.tsx

```diff
@@ -0,0 +1,327 @@
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import {
+  useForm,
+  useFormData,
+  useFormIsModified,
+  Form,
+  fieldFormatters,
+  FormSchema,
+  FIELD_TYPES,
+  UseField,
+  ToggleField,
+  NumericField,
+} from '../../../../../shared_imports';
+
+import { documentationService } from '../../../../services/documentation';
+import { splitSizeAndUnits } from '../../../../../../common';
+import { useAppContext } from '../../../../app_context';
+import { UnitField } from './unit_field';
+import { updateDataRetention } from '../../../../services/api';
+
+interface Props {
+  dataRetention: string;
+  dataStreamName: string;
+  onClose: (data?: { hasUpdatedDataRetention: boolean }) => void;
+}
+
+export const timeUnits = [
+  {
+    value: 'd',
+    text: i18n.translate(
+      'xpack.idxMgmt.dataStreamsDetailsPanel.editDataRetentionModal.timeUnits.daysLabel',
+      {
+        defaultMessage: 'days',
+      }
+    ),
+  },
+  {
+    value: 'h',
+    text: i18n.translate(
+      'xpack.idxMgmt.dataStreamsDetailsPanel.editDataRetentionModal.timeUnits.hoursLabel',
+      {
+        defaultMessage: 'hours',
+      }
+    ),
+  },
+  {
+    value: 'm',
+    text: i18n.translate(
+      'xpack.idxMgmt.dataStreamsDetailsPanel.editDataRetentionModal.timeUnits.minutesLabel',
+      {
+        defaultMessage: 'minutes',
+      }
+    ),
+  },
+  {
+    value: 's',
+    text: i18n.translate(
+      'xpack.idxMgmt.dataStreamsDetailsPanel.editDataRetentionModal.timeUnits.secondsLabel',
+      {
+        defaultMessage: 'seconds',
+      }
+    ),
+  },
+  {
+    value: 'ms',
+    text: i18n.translate(
+      'xpack.idxMgmt.dataStreamsDetailsPanel.editDataRetentionModal.timeUnits.millisecondsLabel',
+      {
+        defaultMessage: 'milliseconds',
+      }
+    ),
+  },
+  {
+    value: 'micros',
+    text: i18n.translate(
+      'xpack.idxMgmt.dataStreamsDetailsPanel.editDataRetentionModal.timeUnits.microsecondsLabel',
+      {
+        defaultMessage: 'microseconds',
+      }
+    ),
+  },
+  {
+    value: 'nanos',
+    text: i18n.translate(
+      'xpack.idxMgmt.dataStreamsDetailsPanel.editDataRetentionModal.timeUnits.nanosecondsLabel',
+      {
+        defaultMessage: 'nanoseconds',
+      }
+    ),
+  },
+];
+
+const configurationFormSchema: FormSchema = {
+  dataRetention: {
+    type: FIELD_TYPES.TEXT,
+    label: i18n.translate(
+      'xpack.idxMgmt.dataStreamsDetailsPanel.editDataRetentionModal.dataRetentionField',
+      {
+        defaultMessage: 'Data retention',
+      }
+    ),
+    formatters: [fieldFormatters.toInt],
+    validations: [
+      {
+        validator: ({ value }) => {
+          if (!value || value === 0) {
+            return {
+              message: i18n.translate(
+                'xpack.idxMgmt.dataStreamsDetailsPanel.editDataRetentionModal.dataRetentionFieldRequiredError',
+                {
+                  defaultMessage: 'A data retention value is required.',
```

Comment:
nit: I think for `value === 0`, I would display a separate message. Maybe even put it together with `value < 0`, something like "a positive value is required".